File: codewars/Permutations/So Many Permutations.py
def permutations(s):
    if len(s) == 1:
        return [s]

    result_set = set()

    for i in range(len(s)):
        char = s[i]
        remaining_chars = s[:i] + s[i+1:]
        remaining_permutations = permutations(remaining_chars)

        for perm in remaining_permutations:
            result_set.add(char + perm)

    return list(result_set)

File: codewars/Permutations/SOMANYPERMUTATIONS.js
function permutations(str) {
    if (str.length <= 1) {
        return [str];
    }

    const resultSet = new Set();

    for (let i = 0; i < str.length; i++) {
        const char = str[i];
        const remainingChars = str.slice(0, i) + str.slice(i + 1);
        const remainingPermutations = permutations(remainingChars);

        for (const perm of remainingPermutations) {
            resultSet.add(char + perm);
        }
    }

    return Array.from(resultSet);
}

const input = "aabb";
const permutationsut = permutations(input);

const formattedOutput = permutationsut.map(permutations => `'${permutations}'`);

console.log(`[${formattedOutput.join(', ')}]`);